src/build/__main__.py

```diff
@@ -181,11 +186,11 @@ def _natural_language_list(elements: Sequence[str]) -> str:
 def build_package(
     srcdir: PathType,
     outdir: PathType,
-    distributions: Sequence[str],
+    distributions: Sequence['Distribution'],
     config_settings: Optional[ConfigSettingsType] = None,
     isolation: bool = True,
     skip_dependency_check: bool = False,
-) -> Sequence[str]:
+) -> List[str]:
```

Comment:
Just following the 'robustness principle' which is espoused by mypy/Python typing. Returning abstract types is fairly unusual.  Do you think list is too restrictive or that it might change in the future?

Reply:
I believe it's better to return well scoped protocols than to return concrete implementations. If we need to change the container here, that would be an API breakage. IMO this is a good practice when designing APIs. There's no reason here why we would need to tie the return value to a list object.
Even in internal APIs, I think limiting the promises is generally good practice, as it prevents us from depending too much on concrete implementations and then having to refactor the code if we need to adjust something.